src/modules/inventory/src/inventoryImp.cpp

```diff
@@ -337,9 +347,19 @@ void Inventory::NotifyChange(ReturnTypeCallback result, const nlohmann::json& da
             msg["operation"] = OPERATION_MAP.at(result);
             msg["data"] = EcsData(data, table);
             msg["id"] = CalculateBase64Id(msg["data"], table);
-            msg["data"]["@timestamp"] = m_scanTime;
-            const auto msgToSend{msg.dump()};
-            m_reportDiffFunction(msgToSend);
+
+            if (msg["id"].is_string() && msg["id"].get<std::string>().size() <= MAX_ID_SIZE)
+            {
+                msg["data"]["@timestamp"] = m_scanTime;
+                const auto msgToSend{msg.dump()};
+                m_reportDiffFunction(msgToSend);
+            }
+            else
+            {
+                LogWarn("Event discarded for exceeding maximum size allowed in id field. Actual: {}, Maximum: {}", msg["id"].get<std::string>().size(), MAX_ID_SIZE);
```

Comment:
```suggestion
                LogWarn("Event discarded for exceeding maximum size allowed in id field.");
```
Do this to avoid having to check again that `id` is a string. In the trace log from below it can be seen the size of the `id`.